src/parser/tptp/tptp.cpp

```diff
@@ -195,6 +199,15 @@ Term TptpState::makeApplyUf(std::vector<Term>& args)
       args[i + 1] = d_solver->mkTerm(TO_REAL, {args[i + 1]});
     }
   }
+  // If a lambda, apply it immediately. This is furthermore important to
+  // avoid lambdas with `-o raw-benchmark` when higher-order is not enabled.
```

Comment:
Wouldn't it be more accurate to leave the lambdas and have the logic of the SMT-LIB problem be HO_...?

Reply:
Also notice that we often introduce lambdas artificially in FOL problems (e.g. https://github.com/cvc5/cvc5/blob/main/src/parser/tptp/Tptp.g#L478).

In theory we could try to distinguish which cases are artificial lambdas, but this is not worth the effort at this point.

Plus, it is worth noting that the above change actually improves our performance a decent amount over all of TPTP.

Reply:
I see. Agreed that better to go this route then.